core/java/iesi-rest-without-microservices/src/test/java/io/metadew/iesi/server/rest/component/dto/ComponentRepositoryDtoTest.java

```diff
@@ -0,0 +1,250 @@
+
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest(classes = Application.class, properties = {"spring.main.allow-bean-definition-overriding=true"})
+@ContextConfiguration(classes = TestConfiguration.class)
+@ActiveProfiles("test")
+@DirtiesContext
+public class ComponentRepositoryDtoTest {
+
+    @Autowired
+    private ComponentDtoRepository componentDtoRepository;
+    @Autowired
+    private ComponentConfiguration componentConfiguration;
+    @Autowired
+    private MetadataRepositoryConfiguration metadataRepositoryConfiguration;
+    @Autowired
+    private ComponentDtoResourceAssembler componentDtoResourceAssembler;
+
+    @AfterEach
+    void cleanup() {
+        metadataRepositoryConfiguration.clearAllTables();
+    }
+
+    @Test
+    void getAllNoResultsTest() {
+        assertThat(componentDtoRepository.getAll(Pageable.unpaged(), new ArrayList<>()))
+                .isEmpty();
+    }
+
+    @Test
+    void getAllTest() {
+        Component component1 = createComponent("component1");
+        Component component2 = createComponent("component2");
+        componentConfiguration.insert(component1);
+        componentConfiguration.insert(component2);
+        ComponentDto component1Dto = componentDtoResourceAssembler.toModel(component1);
+        ComponentDto component2Dto = componentDtoResourceAssembler.toModel(component2);
+        assertThat(componentDtoRepository.getAll(Pageable.unpaged(), new ArrayList<>())).containsOnly(
+                component1Dto,
+                component2Dto
+        );
+    }
+
+    @Test
+    void getAllPaginatedAllInclusiveTest() {
+        Component component1 = createComponent("component1");
+        Component component2 = createComponent("component2");
+        componentConfiguration.insert(component1);
+        componentConfiguration.insert(component2);
+        ComponentDto component1Dto = componentDtoResourceAssembler.toModel(component1);
+        ComponentDto component2Dto = componentDtoResourceAssembler.toModel(component2);
+
+        Pageable pageable = PageRequest.of(0, 2);
+        assertThat(componentDtoRepository.getAll(pageable, new ArrayList<>())).containsOnly(
+                component1Dto,
+                component2Dto
+        );
+    }
+
+    @Test
+    void getAllPaginatedSortedAscTest() {
+        Component component1 = createComponent("component1");
+        Component component2 = createComponent("component2");
+        componentConfiguration.insert(component1);
+        componentConfiguration.insert(component2);
+        ComponentDto component1Dto = componentDtoResourceAssembler.toModel(component1);
+        ComponentDto component2Dto = componentDtoResourceAssembler.toModel(component2);
+
+        Pageable pageable = PageRequest.of(0, 2, Sort.by(Sort.Direction.ASC, "name"));
+        assertThat(componentDtoRepository.getAll(pageable, new ArrayList<>())).containsExactly(
+                component1Dto,
+                component2Dto
+        );
```

Comment:
does 'containsOnly' test order?.

Reply:
Regarding the JUnit documentation, containsOnly doesn't test the order.

Reply:
Regarding the JUnit documentation, containsOnly doesn't test the order.
![Capture](https://user-images.githubusercontent.com/54418529/117233979-6b77c880-ae24-11eb-9d6b-41ab0df4b5c4.PNG)